### PR TITLE
Document kubectl subresource support

### PR DESCRIPTION
**Pull request by nikhita**
Ref: https://github.com/kubernetes/enhancements/issues/2590

### CODE REVIEW

File: content/en/docs/reference/kubectl/conventions.md [hunk 1]
```diff
@@ -19,6 +19,16 @@ For a stable output in a script:
 * Fully-qualify the version. For example, `jobs.v1.batch/myjob`. This will ensure that kubectl does not use its default version that can change over time.
 * Don't rely on context, preferences, or other implicit states.
 
+## Subresources
+
+* You can use the `[--subresource=status|scale]` flag for kubectl commands like `get`, `patch`,
```
Comment (MadhavJivrajani) [inline content/en/docs/reference/kubectl/conventions.md:24]: Would it help to explicitly mention that subresources like `logs` and port forward are not supported similar to how its mentioned in the KEP?

**Review by MadhavJivrajani — COMMENTED** [review]
Thank you! ❤️

File: content/en/docs/reference/kubectl/conventions.md (hunk 1 above)
Comment (nikhita) [reply content/en/docs/reference/kubectl/conventions.md]: Done.

File: content/en/docs/reference/kubectl/conventions.md [hunk 2]
```diff
@@ -19,6 +19,16 @@ For a stable output in a script:
 * Fully-qualify the version. For example, `jobs.v1.batch/myjob`. This will ensure that kubectl does not use its default version that can change over time.
 * Don't rely on context, preferences, or other implicit states.
 
+## Subresources
+
+* You can use the `--subresource` flag for kubectl commands like `get`, `patch`,
```
Comment (soltysh) [inline content/en/docs/reference/kubectl/conventions.md:24]: Nit: maybe worth mentioning this is alpha flag.

**Review by soltysh — APPROVED** [review]
One small nit, but overall lgtm

File: content/en/docs/reference/kubectl/conventions.md (hunk 2 above)
Comment (nikhita) [reply content/en/docs/reference/kubectl/conventions.md]: whoops, not sure how I missed that. Fixed :)

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by reylejano** [comment]
Hi @nikhita , since subresource support is introduced as alpha in v1.24, this PR should target the `dev-1.24` branch, can you change the target branch?
Thank you

**Comment by nikhita** [comment]
@reylejano @sftim base branch fixed.

File: content/en/docs/reference/kubectl/conventions.md [hunk 3]
```diff
@@ -19,6 +19,16 @@ For a stable output in a script:
 * Fully-qualify the version. For example, `jobs.v1.batch/myjob`. This will ensure that kubectl does not use its default version that can change over time.
 * Don't rely on context, preferences, or other implicit states.
 
+## Subresources
+
+* You can use the `--subresource` alpha flag for kubectl commands like `get`, `patch`,
+`edit` and `replace` to fetch and update `status` and `scale` subresources for all resources that
```
Comment (divya-mohan0209) [inline content/en/docs/reference/kubectl/conventions.md:25]:
```suggestion
`edit` and `replace` to fetch and update subresources for all resources
```

File: content/en/docs/reference/kubectl/conventions.md [hunk 4]
```diff
@@ -19,6 +19,16 @@ For a stable output in a script:
...
+support these subresources. Currently, only these two subresources are supported.
```
Comment (divya-mohan0209) [inline content/en/docs/reference/kubectl/conventions.md:26]:
```suggestion
support them. Currently, only the `status` and `scale` subresources are supported.
```

**Review by divya-mohan0209 — COMMENTED** [review]
I've rephrased it a bit to clarify and avoid redundancy. Hope the suggestions are helpful.

**Comment by nikhita** [comment]
@divya-mohan0209 addressed all comments. PTAL.

**Comment by divya-mohan0209** [comment]
Lgtm, thanks!

**Comment by nikhita** [comment]
@sftim can you help lgtm + approve?

File: content/en/docs/reference/kubectl/conventions.md [hunk 5]
```diff
@@ -19,6 +19,16 @@ For a stable output in a script:
 * Fully-qualify the version. For example, `jobs.v1.batch/myjob`. This will ensure that kubectl does not use its default version that can change over time.
 * Don't rely on context, preferences, or other implicit states.
 
+## Subresources
+
+* You can use the `--subresource` alpha flag for kubectl commands like `get`, `patch`,
+`edit` and `replace` to fetch and update subresources for all resources that
+support them. Currently, only the `status` and `scale` subresources are supported.
+* The API contract against a subresource is identical to a full resource. While updating the
+status subresource to hold a new value, keep in mind that the subresource could be potentially
```
Comment (tengqm) [inline content/en/docs/reference/kubectl/conventions.md:28]: nit
```suggestion
`status` subresource to a new value, keep in mind that the subresource could be potentially
```

Comment (nikhita) [reply content/en/docs/reference/kubectl/conventions.md]: @tengqm done.

**Comment by reylejano** [comment]
Preview of updated pages:
https://deploy-preview-32604--kubernetes-io-main-staging.netlify.app/docs/reference/kubectl/cheatsheet/
https://deploy-preview-32604--kubernetes-io-main-staging.netlify.app/docs/reference/kubectl/conventions/

Technical lgtm was provided in [comment](https://github.com/kubernetes/website/pull/32604#pullrequestreview-930461136)

/lgtm
/approve

**Pull request merged**